Add reset reapply to batch reset command

jamipouchi opened the pull request:
<!--- Note to EXTERNAL Contributors -->
<!-- Thanks for opening a PR! 
If it is a significant code change, please **make sure there is an open issue** for this. 
We work best with you when we have accepted the idea first before you code. -->

<!--- For ALL Contributors 👇 -->

## What was changed
<!-- Describe what has changed in this PR -->
Previously reset reapply options were ignored for batch reset, and only used for single workflow reset command.
This pr also adds reset reapply options to batch reset command.

## Why?
<!-- Tell your future self why have you made these changes -->
I have found myself needing to batch-reset workflows without reapplying signals, but unable to do so with the temporal cli. It is possible via sdk code and via tctl, which makes it look as someone forgot to add it.

## Checklist
<!--- add/delete as needed --->

1. Closes <!-- add issue number here -->

2. How was this tested:
<!--- Please describe how you tested your changes/how we can test them -->
I have tested this with my usecase and worked great.
I have not added any tests. Probably some batch-reset tests are needed, but no logic has been added ( I have reused the method that was used for single reset, so there's not much change it's wrong )

3. Any docs updates needed?
<!--- update README if applicable
      or point out where to update docs.temporal.io -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/temporalio/cli?pullRequest=700) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/temporalio/cli?pullRequest=700) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/temporalio/cli?pullRequest=700) it.</sub>

dandavison reviewed (commented):
Nice, thanks a lot @jamipouchi! I made a couple of minor comments but, do you have time to add the test coverage for this?

jamipouchi commented on temporalcli/commands.workflow_reset.go:
```diff
@@ -77,33 +77,9 @@ func (c *TemporalWorkflowResetCommand) doWorkflowReset(cctx *CommandContext, cl
-		}
+	reapplyExcludes, reapplyType, err := getResetReapplyAndExcludeTypes(c.ReapplyExclude.Values, c.ReapplyType.Value)
+	if err != nil {
+		return fmt.Errorf("getting reset reapply and exclude types failed: %w", err)
```
Yup, agree.
Did it cuz it was similar to the error above `getting reset event...` error. I have also changed that one.

jamipouchi commented:
> Nice, thanks a lot @jamipouchi! I made a couple of minor comments but, do you have time to add the test coverage for this?

I am going on vacation tomorrow until next wednesday, and then I'll pick up work.
Happy if anyone can do it for me in the meantime, else I can prob. do it in a couple of weeks.

jamipouchi commented:
> > Nice, thanks a lot @jamipouchi! I made a couple of minor comments but, do you have time to add the test coverage for this?
> 
> I am going on vacation tomorrow until next wednesday, and then I'll pick up work. Happy if anyone can do it for me in the meantime, else I can prob. do it in a couple of weeks.

Hi @dandavison, just added the tests for the batch reset using reapply exclude. Also added first/last workflow task, and one making sure that only wfs matching the query were reset.

dandavison reviewed (approved):
Thanks a lot for this work @jamipouchi, including the careful tests adding coverage that was missing before. My manual testing didn't find any problems.
